### ISSUE TITLE
Importing a class with `\Trait\` in the namespace throws `unexpected 'Trait' (T_TRAIT)`

### ISSUE THREAD

**Issue by robin-vendredi** [glayzzle/php-parser#834]
I stumbled upon an issue while using `@prettier/plugin-php` and I believe it's a bug coming from the `php-parser` lib - please let me know if I'm mistaken. Thank you for your work maintaining `php-parser`.

### What I did

Starting from a fresh install of `3.0.3`, importing any namespace containing `\Trait\` in the path results in a `unexpected 'Trait' (T_TRAIT)` error.

Code to reproduce: 
```js
var parser = require('php-parser');

var eval = parser.parseEval('use App\\Trait\\Foo;');

console.log( 'Eval parse:', eval );
```
throws
```
/code/phpparser/node_modules/php-parser/src/parser.js:345
    throw err;
    ^

SyntaxError: Parse Error : syntax error, unexpected 'Trait' (T_TRAIT), expecting ';' on line 1
    at Parser.raiseError (/code/phpparser/node_modules/php-parser/src/parser.js:337:17)
    at Parser.error (/code/phpparser/node_modules/php-parser/src/parser.js:384:15)
    at Parser.expect (/code/phpparser/node_modules/php-parser/src/parser.js:546:10)
    at Parser.read_use_statement (/code/phpparser/node_modules/php-parser/src/parser/namespace.js:127:10)
    at Parser.read_top_statement (/code/phpparser/node_modules/php-parser/src/parser/statement.js:53:21)
    at Parser.read_start (/code/phpparser/node_modules/php-parser/src/parser/main.js:18:19)
    at Parser.parse (/code/phpparser/node_modules/php-parser/src/parser.js:290:22)
    at Engine.parseEval (/code/phpparser/node_modules/php-parser/src/index.js:146:22)
    at Function.Engine.parseEval (/code/phpparser/node_modules/php-parser/src/index.js:134:15)
    at Object.<anonymous> (/code/phpparser/test.js:4:19) {
  lineNumber: 1,
  fileName: 'eval',
  columnNumber: 8
}
```

Changing the path to `use App\\TraitX\\Foo;` or anything else in the `\Trait\` section of the namespace fixes the error.

### What I expected

Being able to have `Trait` in my namespace without the parser throwing an error.

**Comment by ichiriac** [comment]
The problem comes from the lexer, it differs from PHP lexer here :
https://github.com/php/php-src/blob/master/Zend/zend_language_scanner.l#L2363

Introduced 2 years ago :
https://github.com/php/php-src/commit/7a3dcc3e339cde2177ba4fe8fc45f78c94dfbb29

This issue may apply on any namespace usage (example a catch with a namespace and a token keyword)

@czosel :

A quick fix could be to introduce the `\` here :
https://github.com/glayzzle/php-parser/blob/main/src/lexer/utils.js#L54
=> It may be good, but T_STRING can now have namespaces (so for identifiers like classes or function names you need to return another token in order catch errors)

You could also create this token : T_NAME_QUALIFIED
Implement the parsing branch on the T_STRING :
https://github.com/glayzzle/php-parser/blob/02410c1f9fdc44a3619cb23d4e918fc40ff8c528/src/lexer/tokens.js#L33
=> pseudo algorithm: if next char is `\` then consume_LABEL until is not followed by `\` and return a T_NAME_QUALIFIED token

The T_NAME_FULLY_QUALIFIED could also be implemented but may be more complex to be done

**Comment by czosel** [comment]
Fixed in `3.1.0-beta.6` :tada:
